src/json.hpp.re2c

```diff
@@ -5885,23 +5952,60 @@ class basic_json
 
             case value_t::number_float:
             {
-                // If the number is an integer then output as a fixed with with
-                // precision 1 to output "0.0", "1.0" etc as expected for some
-                // round trip tests otherwise  15 digits of precision allows
-                // round-trip IEEE 754 string->double->string; to be safe, we
-                // read this value from
-                // std::numeric_limits<number_float_t>::digits10
-                if (std::fmod(m_value.number_float, 1) == 0)
+                // Buffer size: precision (2^8-1 = 255) + other ('-.e-xxx' = 7) + null (1)
+                char buf[263];
+                int len;
+
+                if (m_type.bits.parsed)
+                {
+                    // Number was parsed from a string so use the same precision
+                    if (m_type.bits.has_exp)
+                    {
+                        // Exponent - output in exponential form - handle capitalization of e/E
+                        if (m_type.bits.exp_cap)
+                        {
+                            len = snprintf(buf, sizeof(buf), "%.*E", m_type.bits.precision, m_value.number_float) + 1;
+                        }
+                        else
+                        {
+                            len = snprintf(buf, sizeof(buf), "%.*e", m_type.bits.precision, m_value.number_float) + 1;
+                        }
+
+                        // Remove '+' sign from the exponent if necessary
+                        if (!m_type.bits.exp_plus)
+                        {
+                            if (len > static_cast<int>(sizeof(buf))) len = sizeof(buf);
```

Comment:
Hi @twelsby!

I am currently trying to improve the test coverage of the code, but I fail to create a test case where `len` is larger than the size of `buf`. The previous `snprintf` calls never seem to return anything greater than `263` - and if I understand `snprintf` correctly, it would only return a value larger than `sizeof(buf)` if writing to the buffer failed. In that case, capping it to `sizeof(buf)` and continue to write to it would make no sense.

Before I jump to wrong conclusions, I would like to ask you what situation you had in mind that would require the if.

All the best
Niels